gufe/tokenization.py

```diff
@@ -382,7 +399,7 @@ def from_dict(dct) -> GufeTokenizable:
     # was to a deleted object.
     thing = TOKENIZABLE_REGISTRY.get(obj.key)
 
-    if thing is None:
+    if thing is None:  # -no-cov-
```

Comment:
can we test this by saving a dumped object in the testsuite and loading it in a test?

Reply:
> can we test this by saving a dumped object in the testsuite and loading it in a test?

No, this line cannot be tested in the current implementation. It should never be reached; but that's likely to change in #90.

We already do the test you're asking for, by clearing the registry and loading a dumped object (in fact, we do this for *every* object that uses the `GufeTokenizableTestsMixin`).

To see why this doesn't hit that line, note that we can't get `obj.key` in line 400 unless `obj` already exists. Current approach to get the key is to create an object, then get the key from that. On creation, it gets added to the `TOKENIZABLE_REGISTRY`.  So you should have always created an object by here; I actually think this code should be unreachable. This PR affects objects with nondeterministic keys (essentially, random numbers on creation). I think the reason this was hit before was that reloading an object went like this:

1. Create object that should have `key=key1`: on creation it gets `key` that is randomly assigned, call that `key2`. `key1 != key2`. The `TOKENIZABLE_REGISTRY` knows there is an object called `key2`.
2. Set the `key=key1`.
3. Get to line 400 above: `obj.key` gives `key1`, but the registry only has `key2`.

In the new code, step 2 both removes `key2` from the registry and adds `key1`. So we always find an object there.

I'm hoping to change the requirement to recreate the entire object in #90, so that we only generate the key, not a copy of the object. In that case, this line will be reached every time we reload an unknown object.